sdc/tests/tests_perf/test_perf_df_rolling.py

```diff
@@ -0,0 +1,136 @@
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+import string
+import time
+
+import numba
+import pandas
+
+from sdc.tests.test_utils import test_global_input_data_float64
+from sdc.tests.tests_perf.test_perf_base import TestBase
+from sdc.tests.tests_perf.test_perf_utils import (calc_compilation, get_times,
+                                                  perf_data_gen_fixed_len)
+
+
+rolling_usecase_tmpl = """
+def df_rolling_{method_name}_usecase(data):
+    start_time = time.time()
+    res = data.rolling({rolling_params}).{method_name}({method_params})
+    end_time = time.time()
+
+    return end_time - start_time, res
+"""
+
+
+def get_rolling_params(window=100, min_periods=None):
+    """Generate supported rolling parameters"""
+    rolling_params = [f'{window}']
+    if min_periods:
+        rolling_params.append(f'min_periods={min_periods}')
+
+    return ', '.join(rolling_params)
+
+
+def gen_df_rolling_usecase(method_name, rolling_params=None, method_params=''):
+    """Generate df rolling method use case"""
+    if not rolling_params:
+        rolling_params = get_rolling_params()
+
+    func_text = rolling_usecase_tmpl.format(**{
+        'method_name': method_name,
+        'rolling_params': rolling_params,
+        'method_params': method_params
+    })
+
+    global_vars = {'time': time}
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _df_rolling_usecase = loc_vars[f'df_rolling_{method_name}_usecase']
+
+    return _df_rolling_usecase
+
+
+# python -m sdc.runtests sdc.tests.tests_perf.test_perf_df_rolling.TestDFRollingMethods
+class TestDFRollingMethods(TestBase):
+    # more than 19 columns raise SystemError: CPUDispatcher() returned a result with an error set
+    max_columns_num = 19
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.total_data_length = {
```

Comment:
I think this dictionary can move outside of class.

Reply:
@Rubtsowa what advantages do you think you proposal have?

Reply:
Compact code. out of class we can pass all necessary parameters in 1 line.

Reply:
I agree to move it out of the class if it will be used there.